src/compose/app.ts

```diff
@@ -607,29 +698,13 @@ export class App {
 				dependenciesMetForStart,
 				dependenciesMetForKill,
 				needsSpecialKill,
+				currentServicesLocked,
+				currentServices: this.services,
+				appsToLock: context.appsToLock,
 			});
 		}
 	}
 
-	// We return an array from this function so the caller can just concatenate the arrays
-	// without worrying if the step is skipped or not
-	private generateKillStep(
-		service: Service,
-		changingServices: Array<ChangingPair<Service>>,
-	): CompositionStep[] {
-		if (
-			service.status !== 'Stopping' &&
-			!_.some(
-				changingServices,
-				({ current }) => current?.serviceName === service.serviceName,
```

Comment:
This would prevent emitting kill steps for a service that is already stopping. I didn't see this in the new implementation

Reply:
The kill step generation for dependent services got moved to `generateStepsForService`, which already accounts for `Stopping` state by emitting `noop`